src/Microsoft.AspNetCore.SignalR.Redis/RedisHubLifetimeManager.cs

```diff
@@ -242,11 +249,75 @@ public override Task SendUserAsync(string userId, string methodName, object[] ar
             if (connection != null)
             {
                 // short circuit if connection is on this server
-                await AddGroupAsyncCore(connection, groupName);
+                await RemoveGroupAsyncCore(connection, groupName);
                 return;
             }
 
-            await SendGroupActionAndWaitForAck(connectionId, groupName, GroupAction.Add);
+            await SendGroupActionAndWaitForAck(connectionId, groupName, GroupAction.Remove);
+        }
+
+        public override Task SendConnectionsAsync(IReadOnlyList<string> connectionIds, string methodName, object[] args)
+        {
+            if (connectionIds == null)
+            {
+                throw new ArgumentNullException(nameof(connectionIds));
+            }
+
+            var publishTasks = new List<Task>(connectionIds.Count);
```

Comment:
We need to do another pass after this PR to optimize these code paths if they never go async (if that's common)